ui_components/src/travel/Map.js

```diff
@@ -205,102 +283,257 @@ function MapGridLines({ tileWidth, tileHeight, stageOffsetX, stageOffsetY, stage
 function MapLocations({ locations, tileWidth, tileHeight }) {
     return (
         <div className='map_locations'>
-            {locations.map((location) => (
-                <div key={location.location_id}
-                    className={location.objective_type != undefined ? 'map_location ' + location.objective_type : 'map_location'}
+            <ReactTransitionGroup.TransitionGroup>
+                {locations.map((location) => (
+                    <ReactTransitionGroup.CSSTransition
+                        key={location.location_id}
+                        timeout={500} // Set the animation duration in milliseconds
+                        classNames="fade"
+                    >
+                        <div className={location.objective_type != undefined ? 'map_location ' + location.objective_type : 'map_location'}
+                            style={{
+                                cursor: "pointer",
+                                backgroundColor: "#" + location.background_color,
+                                backgroundImage: location.background_image ? `url(${location.background_image})` : null,
+                                transform: location.objective_type == 'key_location' ? `translate3d(${((location.x - 1) * tileWidth) - 8}px, ${((location.y - 1) * tileHeight) - 8}px, 0)`
+                                    : `translate3d(${(location.x - 1) * tileWidth}px, ${(location.y - 1) * tileHeight}px, 0)`,
+                                backfaceVisibility: "hidden",
+                                filter: "blur(0)",
+                            }}>
+                            <div className='map_location_tooltip'>{location.name}</div>
+                            {location.objective_image &&
+                                <div className={location.objective_type != undefined ? 'map_location_objective ' + location.objective_type : 'map_location_objective'}
+                                    style={{
+                                        backgroundImage: "url(." + location.objective_image + ")",
+                                    }}></div>
+                            }
+                        </div>
+                    </ReactTransitionGroup.CSSTransition>
+                ))}
+            </ReactTransitionGroup.TransitionGroup>
+        </div>
+    );
+}
+
+function MapObjectives({ objectives, tileWidth, tileHeight }) {
+    return (
+        <div className='map_objectives'>
+            {objectives.map((objective) => (
+                <div key={objective.id}
```

Comment:
```suggestion
                <div key={`objective:${objective.id}`}
```
I think this way you can have a totally unique ID that avoids the "two elements in different locations have the same numerical ID" trap. I'm not really sure why React pushes for this, but they seem to suggest having truly unique keys?

Reply:
Hmmm... these **should** be totally unique since the ID comes from the database primary key, 
Or is this more of a safety net we should apply to keys in general?